packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue

```diff
@@ -168,145 +166,118 @@ const codeBlock = {
 const codeBlocks = ref([codeBlock]);
 
 onMounted(async () => {
-	const session = await newSession('beaker', 'Beaker');
-	jupyterSession.value = session;
-
-	session.kernelChanged.connect((_context, kernelInfo) => {
-		const kernel = kernelInfo.newValue;
-		if (kernel?.name === 'beaker') {
-			session.iopubMessage.connect(iopubMessageHandler);
-			setKernelContext();
+	try {
+		const jupyterContext = buildJupyterContext();
+		if (jupyterContext) {
+			await kernelManager.init('beaker', 'Beaker', jupyterContext);
 		}
-	});
+	} catch (error) {
+		console.error('Error initializing Jupyter session:', error);
+	}
 });
 
-onUnmounted(async () => {
+onUnmounted(() => {
 	// Get rid of session/kernels on unmount, otherwise we end upwith lots of kernels hanging
 	// around using up memory.
-	jupyterSession.value?.shutdown();
+	kernelManager.shutdown();
 });
 
-function setKernelContext() {
-	const kernel = jupyterSession.value?.session?.kernel;
-	if (!kernel) {
-		return;
-	}
+function buildJupyterContext() {
 	const contextName =
 		selectedModelFramework.value === ModelFramework.Decapodes ? 'decapodes_creation' : null;
 	const languageName =
 		selectedProgrammingLanguage.value === ProgrammingLanguage.Julia ? 'julia-1.9' : null;
```

Comment:
Because those ENUM are set properly, those test should be done in `kernelManager()`

Reply:
Kernel manage helps send the context, it does not create the context. The context creation is up to each individual component.